mmv1/third_party/terraform/resources/resource_compute_instance_group_manager.go.erb

```diff
@@ -317,9 +317,58 @@ func resourceComputeInstanceGroupManager() *schema.Resource {
 				Description: `When used with wait_for_instances specifies the status to wait for. When STABLE is specified this resource will wait until the instances are stable before returning. When UPDATED is set, it will wait for the version target to be reached and any per instance configs to be effective and all instances configs to be effective as well as all instances to be stable before returning.`,
 				<% end -%>
 			},
+			<% unless version == "ga" -%>
+			"stateful_internal_ip": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
```

Comment:
I thought by setting computed and optional to true, we are basically doing the equivalent of default_from_api. I could be wrong as I was referring to https://github.com/GoogleCloudPlatform/magic-modules/blob/5268d03a79ba5644c9b8252175f642245881c8b7/mmv1/overrides/terraform/property_override.rb#L59

Reply:
you are absolutely correct, my bad. That being said, does that API have a default that it sets?

Reply:
The expected value is null for both internal and external ips.

Reply:
if the field is not set in TF, and the api returns null, that sounds like a reasonable return value that shouldn't cause a permadiff - does it? Can we get around it with a customize diff if so?

Basically, if we can avoid making it O+C that's better because we'll be able to accurately report what the end state will be to users (instead of saying `(known after apply)`)

Reply:
I can take a look into a customize diff.

Reply:
Looks like a customize diff is not ideal according to Tf external chat, I will leave it as it is.

Reply:
just to have it recorded here - customize diff / diff suppress should be fine for this case since it would only be handling the empty return value; this is not a field that is partially managed by the API.

Reply:
For Region instance manager, the log: https://paste.googleplex.com/4890762578755584, looks like the current error I'm fighting against. I'll try the other one

Reply:
ok after fixing the error for region instance manager, it looks like I'm still getting 
stateful_external_ip {
  |           - delete_rule    = "NEVER" -> null
  |           - interface_name = "nic0" -> null
  |         }

Reply:
interesting - I see that in paste/5070583447420928 the POST that first creates the instance group already includes values in the `externalIPs` and `internalIPs` fields.

That implies that they're in the Terraform configuration, since they're being sent on Create - but then the test fails with what looks like: a permadiff from those fields being set by the API but _not_ in the terraform configuration. Why is that happening?

Reply:
Good point. This error also happens before the DELETE operation. 
Can we know whether the IPs are stored in the TF state file?

Reply:
it's not easy to do from the tests; you might need to set up a local directory where you can set up files and apply them and then examine the results.

See https://googlecloudplatform.github.io/magic-modules/#using-released-terraform-binary-with-local-provider-binary for how to build & use the provider binary locally.

Reply:
Sure I can take a try.

Reply:
Pushed a draft commit with fixed tests and additional log.print(). 
There is an additional null patching step that caused some problem. 

From what I researched, a customize diff is already implemented in place. Perhaps some details are not right at the moment.

Reply:
Analyzed deeper into this, the issue is that during the step 5 of the test, a patch request with
"statefulPolicy": {
  "preservedState": {}
 }
is sent to the API. However the stateful_ips and the stateful_disk are not removed, therefore the API returns a value whereas TF plan sends a null value.